### DISCUSSION TITLE
DSP Reports API error

### DISCUSSION THREAD

**Discussion by DanielaZamoraUribe** [amzn/ads-advanced-tools-docs#15] [Q&A] [answered]
Hi there!

I have been working on implementing the API connection for Sponsored Ads and DSP in the past couple of days. I have successfully made the connection for Sponsored Ads, but unfortunately I have not succeeded with DSP so I would like to know if I am doing something wrong. I got all the advertiser IDs for my DSP account, and now I am trying to retrieve a report using `/accounts/{accountId}/dsp/reports`, where my account ID is one of the advertiser IDs that I got after calling `/dsp/advertisers`. When I made the POST call to create a report, I got a `406 Not Acceptable` response and no further details.

My headers look like this:

```
headers = {
            "Authorization": f"Bearer {access_token}",
            "Amazon-Advertising-API-ClientId": ADS_API_CLIENT_ID,
        }
```

And the complete url looks like this: `https://advertising-api-eu.amazon.com/accounts/advertiserId/dsp/reports/` including those headers mentioned above.

I also noticed that have a manager account that has all the DSP and Sponsored Ads accounts together, so I tried using its entity ID as the paramater for `accountId`, which would look like `amzn1.ads1.ma1.XXXXXXXXXXXXXXXXXXX`, but then I got the error `401 Not authorized`. Do you know what could be wrong here?.

Thanks in advance for your help!

**Comment by DanielaZamoraUribe** [comment, marked as answer]
Fixed. Closing this.

**Comment by takeno-dev** [comment]
@DanielaZamoraUribe 
I have the same problem. Please let me know how to solve it.

**Comment by takeno-dev** [comment]
"Accept: application/vnd.dspcreatereports.v3+json"
is attached but 406 is returned.

**Comment by takeno-dev** [comment]
Resolved!

**Comment by kdcurrie** [comment]
@DanielaZamoraUribe
I'm now receiving a 401 Not authorized when trying to generate DSP reports. I've been able to generate reports in the past and have not changed my postman environment configuration.

I'm using an accountId of type "agency"
url: https://advertising-api.amazon.com/accounts/accountId/dsp/reports/

with these headers
![Screen Shot 2023-02-24 at 1 28 21 PM](https://user-images.githubusercontent.com/42992348/221296587-46098642-93f7-45df-94c2-4078f7edf97e.png)

**Reply by elizafarley** [reply]
@kdcurrie A couple suggestions to troubleshoot:

1. If you are using GET /v2/profiles to retrieve your agency profiles, make sure you are using `accountInfo.id` as your DSP `accountId` in the path. 
2. If you are specifying any `advertiserIds` in the request body, make sure you have access to all those advertisers. As a test you could try making a request without specifying any `advertiserIds` and see if you still get an authorization error.

**Reply by kdcurrie** [reply]
Resolved! I believe there was an issue where profileId was being used instead of accountInfo.id before. It seems that issue has been fixed and now accountInfo.id should be used as accountId in the path. Thanks for the help!
